lightning/src/routing/network_graph.rs

```diff
@@ -1123,6 +1133,64 @@ impl NetworkGraph {
 		self.update_channel_from_unsigned_announcement_intern(msg, None, chain_access)
 	}
 
+	/// Update channel from partial announcement data received via rapid gossip
+	///
+	/// `timestamp: u64`: Timestamp the announcement was originally received.
+	///
+	/// All other parameters as used in [`msgs::UnsignedChannelAnnouncement`] fields.
+	pub fn update_channel_from_partial_announcement(&self, short_channel_id: u64, timestamp: u64, features: ChannelFeatures, node_id_1: PublicKey, node_id_2: PublicKey) -> Result<(), LightningError> {
```

Comment:
Should this be called something like `add_channel_from_partial_announcement` instead?

Reply:
the other from_announcement method is called update, so I figured I'd leave it consistent, but I don't have any objections either way.